src/librustc_interface/util.rs

```diff
@@ -240,8 +240,9 @@ pub fn get_codegen_backend(sess: &Session) -> Box<dyn CodegenBackend> {
             .debugging_opts
             .codegen_backend
             .as_ref()
-            .unwrap_or(&sess.target.target.options.codegen_backend);
-        let backend = match &codegen_name[..] {
+            .map(|backend| &backend[..])
```

Comment:
Option::as_deref should remove one line of code.

Reply:
Thanks for the tip!